build.sh

```diff
@@ -1,27 +1,26 @@
 #!/bin/bash
 
-version="$1"
-mkdir -p "binairies/""$version"
-name="binairies/""$version""/linx-client-v""$version""_"
-
-GOOS=darwin GOARCH=amd64 go build -o "$name"osx-amd64
-
-GOOS=darwin GOARCH=386 go build -o "$name"osx-386
-
-GOOS=freebsd GOARCH=amd64 go build -o "$name"freebsd-amd64
-
-GOOS=freebsd GOARCH=386 go build -o "$name"freebsd-386
-
-GOOS=openbsd GOARCH=amd64 go build -o "$name"openbsd-amd64
-
-GOOS=openbsd GOARCH=386 go build -o "$name"bsd-386
-
-GOOS=linux GOARCH=arm go build -o "$name"linux-arm
-
-GOOS=linux GOARCH=amd64 go build -o "$name"linux-amd64
-
-GOOS=linux GOARCH=386 go build -o "$name"linux-386
-
-GOOS=windows GOARCH=amd64 go build -o "$name"windows-amd64.exe
-
-GOOS=windows GOARCH=386 go build -o "$name"windows-386.exe
+if [ -n "$1" ]; then
+	version="$1"
+else
+	version=$(git tag | grep ^v | tail -1 | sed 's/^v//')
+fi
+
+echo "Building version ${version}..."
+
+bin_dir="binaries/$version"
+mkdir -p "$bin_dir"
+
+build() {
+	echo "Building for $1-${2}..."
+	env GOOS="$1" GOARCH="$2" go build \
+	  -o "binaries/$version/linx-client-v${version}_$1-$2"
+}
+
+for os in darwin freebsd openbsd linux windows; do
+	for arch in amd64 386; do
+		build "$os" "$arch"
+	done
+done
+
+build linux arm
```

Comment:
No, "build linux arm" was outside the loop because none of the other archs were being built for arm.